website/static/js/contribRemover.js

```diff
@@ -0,0 +1,275 @@
+                    var node = self.nodesOriginal()[key];
+                    var contributors = node.contributors;
+                    for (var i = 0; i < contributors.length; i++) {
+                        if (contributors[i].id === self.contributorToRemove().id) {
+                            titlesToRemove.push(node.title);
+                            break;
+                        }
+                    }
+                }
+            }
+            return titlesToRemove;
+        });
+
+        self.titlesToKeep = ko.computed(function() {
+            var titlesToKeep = [];
+            for (var key in self.nodesOriginal()) {
+                if (self.nodesOriginal().hasOwnProperty(key) && !self.canRemoveNodes()[key]) {
+                    var node = self.nodesOriginal()[key];
+                    var contributors = node.contributors;
+                    for (var i = 0; i < contributors.length; i++) {
+                        if (contributors[i].id === self.contributorToRemove().id) {
+                            titlesToKeep.push(node.title);
+                            break;
+                        }
+                    }
+                }
+            }
+            return titlesToKeep;
+        });
+
+        self.nodeIDsToRemove = ko.computed(function() {
+            var nodeIDsToRemove = [];
+            for (var key in self.nodesOriginal()) {
+                if (self.nodesOriginal().hasOwnProperty(key) && self.canRemoveNodes()[key]) {
+                    var node = self.nodesOriginal()[key];
+                    var contributors = node.contributors;
+                    for (var i = 0; i < contributors.length; i++) {
+                        if (contributors[i].id === self.contributorToRemove().id) {
+                            nodeIDsToRemove.push(node.id);
+                            break;
+                        }
+                    }
+                }
+            }
+            return nodeIDsToRemove;
+        });
+
+        $.ajax({
+            url: window.contextVars.node.urls.api + 'tree',
+            type: 'GET',
+            dataType: 'json'
+        }).done(function(response) {
+            nodesOriginal = $osf.getNodesOriginal(response[0], nodesOriginal);
+            self.nodesOriginal(nodesOriginal);
+        }).fail(function(xhr, status, error) {
+            $osf.growl('Error', 'Unable to retrieve projects and components');
+            Raven.captureMessage('Unable to retrieve projects and components', {
+                url: self.nodeApiUrl, status: status, error: error
+            });
+        });
+    },
+    clear: function() {
+        var self = this;
+        self.deleteAll(false);
+    },
+    back: function() {
+        var self = this;
+        self.page(self.REMOVE);
+    },
+    submit: function() {
+        var self = this;
+        var removeUrl = window.contextVars.node.urls.api + 'contributor/remove/';
+        $osf.postJSON(
```

Comment:
Methods that do async work should return promises. In this case, you can just return the return value of `postJSON`. This allows for callers to do useful things with the promise

```js
vm.submit().then(doSomethingCoolWithResponse);
```